google/cloud/bigquery/client.py

```diff
@@ -3221,9 +3222,31 @@ def query(
 
         job_ref = job._JobReference(job_id, project=project, location=location)
         query_job = job.QueryJob(job_ref, query, client=self, job_config=job_config)
-        query_job._begin(retry=retry, timeout=timeout)
 
-        return query_job
+        try:
+            query_job._begin(retry=retry, timeout=timeout)
+        except core_exceptions.GoogleAPICallError as create_exc:
```

Comment:
Wouldn't it make sense only to catch `core_exceptions.Conflict` here?  Other errors presumably can't involve an already-created job.

Reply:
If I understood correctly, we want to catch any exception here, because even if one occurs, a job might have nevertheless been created. Admittedly, that's weird to me.

Reply:
Makes sense, I changed the logic to only try to recover from Conflict errors.